app/sidekiq/simple_forms_api/form_remediation/upload_retry_job.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module SimpleFormsApi
+  module FormRemediation
+    class UploadRetryJob
+      include Sidekiq::Job
+
+      sidekiq_options retry: 10
+
+      STATSD_KEY_PREFIX = 'api.simple_forms_api.upload_retry_job'
+
+      sidekiq_retries_exhausted do |_msg, ex|
+        StatsD.increment("#{STATSD_KEY_PREFIX}.retries_exhausted")
+        Rails.logger.error(
+          'SimpleFormsApi::FormRemediation::UploadRetryJob retries exhausted',
+          { exception: "#{ex.class} - #{ex.message}", backtrace: ex.backtrace&.join("\n").to_s }
+        )
+      end
+
+      def perform(file, directory, config)
+        @file = file
+        @directory = directory
+        @config = config
+        uploader = config.uploader_class.new(directory:, config:)
+
+        begin
+          StatsD.increment("#{STATSD_KEY_PREFIX}.total")
+
+          uploader.store!(file)
+        rescue Aws::S3::Errors::ServiceError
+          raise if service_available?(config.s3_settings.region)
+
+          retry_later
+        end
+      end
+
+      private
+
+      attr_accessor :file, :directory, :config
+
+      def service_available?(region)
+        Aws::S3::Client.new(region:).list_buckets
```

Comment:
Are we using `list_buckets` as a proxy for determining the health of the endpoint? Does S3 not provide a more direct way to know if the service is available?

Reply:
So, this is something I did a bit of research on and from what I can tell, there isn't as straightforward an approach to determine AWS S3 health as doing this. This approach leverages the Ruby AWS SDK itself which to me makes the most sense given it's what we're utilizing everywhere else in the app. To my knowledge, this gem doesn't provide direct health checks other than just querying S3 buckets like so. AWS's documentation is overly verbose and widespread in my opinion so a different solution most likely exists but I decided on this. I'm totally down to implement a different/better option if you have something in mind or can point me towards where we're checking elsewhere but I think this will be sufficient for what we're attempting to accomplish.